pkg/securitycontextconstraints/seccomp/strategy.go

```diff
@@ -0,0 +1,176 @@
+package seccomp
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	podutil "k8s.io/kubernetes/pkg/api/pod"
+	api "k8s.io/kubernetes/pkg/apis/core"
+)
+
+const (
+	allowAnyProfile = "*"
+)
+
+// Strategy defines the interface for all seccomp constraint strategies.
+type SeccompStrategy interface {
+	// Generate returns a profile based on constraint rules.
+	Generate(annotations map[string]string, pod *api.Pod) (string, error)
+	// Validate ensures that the specified values fall within the range of the strategy.
+	ValidatePod(pod *api.Pod) field.ErrorList
+	// Validate ensures that the specified values fall within the range of the strategy.
+	ValidateContainer(pod *api.Pod, container *api.Container) field.ErrorList
+}
+
+type strategy struct {
+	allowedProfiles []string
+	// does the strategy allow any profile (wildcard)
+	allowAnyProfile       bool
+	runtimeDefaultAllowed bool
+}
+
+var _ SeccompStrategy = &strategy{}
+
+// NewStrategy creates a new strategy that enforces seccomp profile constraints.
+func NewSeccompStrategy(allowedProfiles []string) SeccompStrategy {
+	allowAny := false
+	allowed := make([]string, 0, len(allowedProfiles))
+	runtimeDefaultAllowed := false
+
+	for _, p := range allowedProfiles {
+		if p == allowAnyProfile {
+			allowAny = true
+			continue
+		}
+		// With the graduation of seccomp to GA we automatically convert
```

Comment:
I'm just trying to understand intent, not do a deep review.  Is this conversion of docker/default to runtime/default new?

Reply:
Apparently not that super new - it seems to have been introduced in 1.19: https://github.com/kubernetes/kubernetes/issues/95817 - along with the seccomp profiles reaching GA in that release.

Reply:
can we add that link as a comment for future reference?

Reply:
I don't add links to code, I may add it to the commit though.

Reply:
I only added the information about the change happening in 1.19 in the commit message.